netflix-roulette/src/hooks/useFilteredMovies.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+import { GenreType } from "@utils/constants";
+import { useHomePageData } from "@data/HomePageDataProvider";
+
+export const useFilteredMoviesList = () => {
+  const { searchText, movies, selectedGenre } = useHomePageData();
+  const shouldPerformSearch =
+    searchText.trim().length !== 0 || selectedGenre !== GenreType.ALL;
+  const searchMoviesByTextAndGenre = () => {
+    return movies.filter(
+      (movie) =>
+        movie.title
+          .toLowerCase()
+          .includes(searchText.trimStart().toLowerCase()) &&
+        (selectedGenre === GenreType.ALL ||
+          movie.genres.includes(selectedGenre))
+    );
+  };
+
+  return React.useMemo(
```

Comment:
I believe, that may have some unexpected side-effect, like, when you modify some arbitrary movie record (e.g. change movie title) you still may get  memoized search results (ignoring most recent changes).